test/e2e/network/dns_common.go

```diff
@@ -622,7 +638,7 @@ func generateDNSUtilsPod() *v1.Pod {
 			Containers: []v1.Container{
 				{
 					Name:    "util",
-					Image:   imageutils.GetE2EImage(imageutils.Dnsutils),
+					Image:   imageutils.GetE2EImage(imageutils.Agnhost),
```

Comment:
There isn't. We could create one, but the way an agnhost pod varies quite a bit, including the command, args, volumes attached, and in some cases, restart policy, readiness probes, env vars, and so on. It would be a bit more difficult to create such a method to be used in such various ways, and it would make these Agnhost replacement PRs harder due to the increased LoC churn.

Reply:
I get that, but there's nothing "dns-utils" about the pod returned from this function now, other than its name. The command ("sleep") is pretty generic, and none of the other things you mentioned get set by this function.

I mean, actually, there *is* a function you can use to replace this one; `e2epod.NewExecPodSpec()`

Reply:
Yeah, that makes sense. Will do that then.

I've also looked into creating a more generic ``NewAgnhostPod`` function that could be used. It's still WIP at the moment: https://github.com/kubernetes/kubernetes/pull/81226/commits/de7ea7946e6bbced9bc3ef5b9185d182298c0135